### PR TITLE
feat: improve CheckBox state management and RadioButtonManager integration

### PR DESCRIPTION
**Pull request by MasatoMakino**
## Summary
• Fix ButtonInteractionHandler press state management issues identified by CodeRabbit
• Implement API separation between public (state-preserving) and internal (state-forcing) selection methods
• Resolve RadioButtonManager material state inconsistencies with comprehensive testing

### State Management Improvements
- **Proactive state clearing**: Added early clearing of press/hover states on disable/freeze operations
- **Public selection API**: Preserves current interaction state (hover, disable) for consistent UX
- **Internal override API**: Forces normal state for predictable RadioButtonManager behavior

### API Design Enhancement  
- **`selection` setter**: Uses `updateMaterial()` to preserve interaction state, respects disable/frozen
- **`_setSelectionOverride()`**: Uses `updateState("normal")` for destructive changes, bypasses validation
- **Clear separation**: Public API for user interactions, internal API for manager systems

### RadioButtonManager Integration Fix
- **Material consistency**: Ensures `normalSelect` material even when selecting hover-state buttons
- **Frozen behavior**: Selected buttons maintain `normalSelect` appearance regardless of interaction state
- **Integration testing**: Added 3 comprehensive test cases covering material state scenarios

## Test Coverage
- **210 total tests** (all passing)
- **New RadioButtonManager tests**: Material state integration with hover/frozen scenarios  
- **False positive verification**: Tests correctly detect the original issue when introduced
- **Comprehensive validation**: All state transitions and API behaviors covered

### Problem Analysis
1. **Stale press state**: Buttons retained press state after disable/freeze cycles
2. **Selection API conflicts**: Public safety vs RadioButtonManager override needs
3. **Material inconsistency**: Hover-state buttons showed `overSelect` instead of `normalSelect`

### Solution Implementation
1. **Proactive clearing**: `switchEnable()` and `frozen` setter clear transient states
2. **API separation**: Distinct methods for user interactions vs manager operations  
3. **Material forcing**: Internal API ensures predictable visual appearance

### Quality Assurance
- **Comprehensive testing**: Edge cases, state transitions, integration scenarios
- **Documentation accuracy**: JSDoc updated to match implementation behavior
- **Code quality**: All lint issues resolved, TypeScript strict compliance

## Test Plan
- [x] All existing tests pass (210/210)
- [x] New RadioButtonManager integration tests verify material behavior
- [x] False positive testing confirms issue detection capability
- [x] Build and documentation generation successful
- [x] Code quality checks (Biome, TypeScript) pass

🤖 Generated with [Claude Code](https://claude.ai/code)

## Summary by CodeRabbit

* Bug Fixes
  * Selection respects disabled/exclusive states and preserves visual state across disable/enable, freeze/unfreeze, and hover transitions.
  * Programmatic selection changes do not emit user-selection events; programmatic setters respect activity checks.

* New Features
  * Backward-compatible checkbox alias class added (deprecation warning).
  * Public exclusivity flag and a _setSelectionOverride hook for radio items to support forced selection behavior.

* Refactor
  * Selection logic reworked to use explicit exclusivity/override semantics; docs clarified.

* Tests
  * New comprehensive tests for checkbox and radio selection, exclusivity, hover/disable interactions, material-state transitions, and redundant-update optimization.

### CODE REVIEW

File: src/interactionHandler/CheckBoxInteractionHandler.ts [hunk 1]
```diff
@@ -121,14 +121,40 @@ export class CheckBoxInteractionHandler<
...
+   * forces transition to "normal" state for consistent visual appearance.
+   *
+   * @internal
+   * @remarks
+   * - This method is intended only for internal library use
+   * - External application code should use the public selection setter which respects object state
+   * - Forces "normal" state transition to ensure predictable material appearance (e.g., normalSelect)
+   * - Used when exclusive selection systems need to override user interaction states
+   */
+  public _setSelectionOverride(bool: boolean): void {
```
[the remark this answers is not in the document]
Comment (MasatoMakino) [reply src/interactionHandler/CheckBoxInteractionHandler.ts:157]: f27e76d646117adcc8147c2de1d02ddea36c21c2
で対応。

File: src/RadioButtonManager.ts [hunk 2]
```diff
@@ -206,7 +206,8 @@ export class RadioButtonManager<Value = unknown> extends EventEmitter<
 
     this._selected = interactionHandler;
     for (const mdl of this._interactionHandlers) {
-      mdl.selection = mdl.isFrozen = mdl === interactionHandler;
+      mdl.isFrozen = mdl === interactionHandler;
+      mdl._setSelectionOverride(mdl === interactionHandler);
     }
```
[the remark this answers is not in the document]
Comment (MasatoMakino) [reply src/RadioButtonManager.ts:211]: f27e76d646117adcc8147c2de1d02ddea36c21c2
で推奨案を反映。
